src/mqt/problemsolver/partialcompiler/Partial_QAOA.py

```diff
@@ -0,0 +1,192 @@
+
+        mapping_fct = self.compile_with_mapping if consider_mapping else self.compile_without_mapping
+        qc_prep_compiled = mapping_fct(qc_prep)
+        qcs_problem_compiled = []
+        qcs_mixer_compiled = []
+        for i in range(len(qcs_problem_uncompiled)):
+            qcs_problem_compiled.append(mapping_fct(qcs_problem_uncompiled[i]))
+            qcs_mixer_compiled.append(mapping_fct(qcs_mix_uncompiled[i]))
+
+        return qc_prep_compiled, qcs_problem_compiled, qcs_mixer_compiled
+
+    def compile_without_mapping(self, qc: QuantumCircuit, opt_level: int = 3) -> QuantumCircuit:
+        return transpile(
+            qc,
+            basis_gates=IBM_GATES,
+            optimization_level=opt_level,
+        )
+
+    def compile_with_mapping(self, qc: QuantumCircuit, opt_level: int = 3) -> QuantumCircuit:
+        return transpile(
+            qc,
+            basis_gates=IBM_GATES,
+            initial_layout=self.mapping,
+            coupling_map=self.coupling_map,
+            layout_method="trivial",
+            optimization_level=opt_level,
+        )
+
+    def get_uncompiled_fully_composed_circuit(self) -> QuantumCircuit:
+        """
+        Returns the fully composed circuit without any compilation.
+        """
+        qc_prep, qcs_problem, qcs_mix = self.get_uncompiled_circuit(include_online_edges=True)
+        qc = qc_prep
+        for i in range(len(qcs_problem)):
+            qc.compose(qcs_problem[i], inplace=True)
+            qc.compose(qcs_mix[i], inplace=True)
+        return qc
+
+    def compile_full_circuit(self) -> QuantumCircuit:
+        """
+        Returns the fully composed circuit compiled to the selected quantum device.
+        """
+        qc = self.get_uncompiled_fully_composed_circuit()
+        return transpile(
+            qc,
+            coupling_map=self.coupling_map,
+            basis_gates=IBM_GATES,
+            optimization_level=1,
+        )
+
+    def get_uncompiled_online_edges(self) -> list[QuantumCircuit]:
+        """
+        Returns the online edges uncompiled for all repetitions.
+        """
+        qc_online_edges_uncompiled_all_reps = []
+        for i in range(self.repetitions):
+            qc_online_edges = QuantumCircuit(self.num_qubits)
+            for elem in self.online_time_edges:
+                qc_online_edges.rzz(self.problem_parameters[i], elem[0], elem[1])
+            qc_online_edges_uncompiled_all_reps.append(qc_online_edges)
+
+        return qc_online_edges_uncompiled_all_reps
+
+    def get_compiled_online_edges(self) -> list[QuantumCircuit]:
+        """
+        Returns the online edges compiled for all repetitions.
+        """
+        assert self.mapping
+        qc_online_edges_compiled_all_reps = []
+        qc_online_edges_uncompiled_all_reps = self.get_uncompiled_online_edges()
+        for qc in qc_online_edges_uncompiled_all_reps:
+            qc_online_edges_compiled_all_reps.append(self.compile_with_mapping(qc, opt_level=1))
```

Comment:
Generator expressions could be also used here.

Reply:
Implemented that.